Yosemite/Yosemite/Stores/ProductTagStore.swift

```diff
@@ -96,6 +88,9 @@ private extension ProductTagStore {
     /// Create new product tags associated with a given Site ID.
     ///
     func addProductTags(siteID: Int64, tags: [String], onCompletion: @escaping (Result<[ProductTag], Error>) -> Void) {
+        guard tags.isEmpty == false else {
+            return onCompletion(.success([]))
+        }
```

Comment:
I found it interesting that we need this check here, after checking the code, the issue is in the handling of tags in iOS, the logic in `ProductTagsViewController` could use some optimization, as currently it always tries to create tags even the already added ones, and by extension causes the issue that you noticed.
Normally, the logic should be create just the new tags, and would speed up the process also (when the user picks just existing ones).

I'll create an issue to keep track of this improvement; for now, this fix here should be enough.

Reply:
Thank you!

Reply:
Since this is a simple fix, I sneaked it in c77d166c09054466c27e2937c0b708024be077d2.

Reply:
Thanks @itsmeichigo for the additional improvement, this looks good, but my remark touches another point, currently the logic of iOS always triggers an API request to create the tags, but this should be needed only when the user adds a new one that doesn't already exist, when the user picks tags from the suggestions, we don't need to re-create them. The following screen recording shows what I mean (I'm selecting an existing tag, and yet we show a loading indicator and we send an API request):

https://github.com/user-attachments/assets/4e57e9f3-c68f-4d25-8688-9796eff4b9e7

I created this ticket https://github.com/woocommerce/woocommerce-ios/issues/14555 to keep track of this.